### PR TITLE
Enable network Select-String test

### LINKED ISSUE #4827: Enable Select-String test with network path
We have the skipped [Select-String test](https://github.com/PowerShell/PowerShell/blob/02b5f357a20e6dee9f8e60e3adb9025be3c94490/test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1#L39)
```powershell
        It "Network path" -skip:($IsCoreCLR) {
            (select-string -LiteralPath $fileNameAsNetworkPath "b").count | Should Be 2
        }
```

Should we enable the test or remove?

If enable we should:
1. ~Run it with admin privileges~
2. Fix #4828  `New-TemporaryFile` - it don't return extended provider properties like PSDrive, PSPath...

### PR DESCRIPTION
**Pull request by iSazonov**
Fix #4827 

Enable pending network Select-String test.

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1 [hunk 1]
```diff
@@ -14,7 +14,8 @@
             $pathWithoutWildcard = $TestDrive
             $pathWithWildcard = Join-Path $TestDrive '*'
 
-            $tempFile = New-TemporaryFile
+            # Here Get-ChildItem add 'PSDrive' property
```
Comment (markekraus) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1]: I had trouble understanding this comment. Can you correct `add` to `adds`?

**Review by markekraus — APPROVED** [review]
Minor comment suggestion, otherwise LGTM

File: test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1]: Fixed.

**Review by markekraus — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1 [hunk 2]
```diff
@@ -14,7 +14,8 @@
             $pathWithoutWildcard = $TestDrive
             $pathWithWildcard = Join-Path $TestDrive '*'
 
-            $tempFile = New-TemporaryFile
+            # Here Get-ChildItem adds 'PSDrive' property
+            $tempFile = New-TemporaryFile | Get-ChildItem
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1]: Although in this case, it'll have the same result, I think

```powershell
New-TemporaryFile | Get-Item
```

is more correct, particularly if other people use this pattern until we address `New-TemporaryFile`

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1 (hunk 2 above)
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/string.tests.ps1]: Fixed.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**